src/GoogleReader.m

```diff
@@ -51,27 +49,31 @@
 
 enum GoogleReaderStatus {
 	notAuthenticated = 0,
-	isAuthenticating,
-	isMissingToken,
-	isGettingToken,
-	isAuthenticated
+	waitingClientToken,
+	missingTToken,
+	waitingTToken,
+	fullyAuthenticated
 } googleReaderStatus;
 
 @interface GoogleReader()
 @property (nonatomic, copy) NSMutableArray * localFeeds;
-@property (atomic, copy) NSString *token;
+@property (atomic, copy) NSString *tToken;
 @property (atomic, copy) NSString *clientAuthToken;
-@property (nonatomic, strong) NSTimer * tokenTimer;
-@property (nonatomic, strong) NSTimer * authTimer;
+@property (nonatomic, strong) NSTimer * tTokenTimer;
+@property (nonatomic, strong) NSTimer * clientAuthTimer;
+@property (nonatomic, copy) NSMutableArray * clientAuthWaitQueue;
+@property (nonatomic, copy) NSMutableArray * tTokenWaitQueue;
 @end
 
 @implementation GoogleReader
 
 @synthesize localFeeds;
-@synthesize token;
+@synthesize tToken;
 @synthesize clientAuthToken;
-@synthesize tokenTimer;
-@synthesize authTimer;
+@synthesize tTokenTimer;
```

Comment:
Oh I should have said this earlier, but properties are auto-synthesized now. We can clean it up in the future